### DISCUSSION TITLE
Upload File

### DISCUSSION THREAD

**Discussion by Shruti3004** [fossology/FOSSologyUI#43] [Q&A] [answered]
@GMishx I am not able to understand the use of `optional Analysis`, `concluded License Decider` and `Reuse` I have created them but these are fields are not in api so how they will affect the upload of a file.

![Screenshot from 2021-06-28 09-12-25](https://user-images.githubusercontent.com/56133783/123577199-fe176100-d7f0-11eb-85d4-1e3960825861.png)

**Comment by GMishx** [comment, marked as answer]
From REST API, upload and scan are two different processes. First the file needs to be uploaded and in response, you will get the upload id
https://github.com/fossology/fossology/wiki/FOSSology-REST-API#2-uploading-a-new-package

Then a second call needs to be done on that upload ID to trigger the scans: https://github.com/fossology/fossology/wiki/FOSSology-REST-API#4-schedule-scan-agents

**Reply by GMishx** [reply]
@Shruti3004 I think the PR #46 (merged) included these fields already. Do you still need some more clarification on these?

**Comment by Shruti3004** [comment]
No @GMishx, Thanks for the help.

**Comment by Shruti3004** [comment]
@GMishx 
https://github.com/fossology/fossology/wiki/FOSSology-REST-API#2-uploading-a-new-package in this we are not having the field for highlighted parameters


![Screenshot from 2021-07-12 09-45-49](https://user-images.githubusercontent.com/56133783/125236246-43fc1b00-e301-11eb-8f15-4e1e83b1cedb.png)

**Reply by Shruti3004** [reply]
For comma separated lists and recursion depth.

**Reply by GMishx** [reply]
Please create an issue for the same.

**Reply by Shruti3004** [reply]
Okay @GMishx and from where I can check the parameters for Upload from server it is not listed in the documentation.

**Reply by Shruti3004** [reply]
@GMishx Opened the issue for same [https://github.com/fossology/fossology/issues/2050](https://github.com/fossology/fossology/issues/2050)
